packages/edit-post/src/components/header/index.js

```diff
@@ -87,6 +90,9 @@ function Header( {
 		};
 	}, [] );
 
```

Comment:
`POST_TYPE_EDITOR_INTERFACE` is not structured quite right as we don't want the header bar and surround in call cases, currently only if linking from within a full post, but not if linking from pattern list in wp_admin. maybe it needs to be something like:
```
export const POST_TYPE_EDITOR_INTERFACE = {
    focusedMode {	
         wp_block: {
		hasDocumentBar: true,
		hasBackLink: true,
		hasSurround: true,
	},
    }
};
```
the check can then be `POST_TYPE_EDITOR_INTERFAC[editMode][ postType ]?.hasDocumentBar. That is pretty messy though, we probably need to abstract this to a method or selector if adding that extra layer.

Reply:
I wasn't aware of that.

An alternative is that it works on the basis of whether there's any router history and then get rid of the `POST_TYPE_EDITOR_INTERFACE` constant. 

So:
- Editing post and no history - `documentBar: false`, `hasBackLink: false`, `hasSurround: false`
- Editing pattern and no history - `documentBar: false`, `hasBackLink: false`, `hasSurround: false`
- Editing anything else when there's history - `documentBar: true`, `hasBackLink: true`, `hasSurround: true`

It certainly seems like that's how the back link should work, though I'm not as certain about what the rules are supposed to be for the other parts of the UI.

Had a quick look at `@wordpress/router` and it seems like it should be possible, but it isn't something that the history package supports out of the box. I think `history.listen` would need to be exposed and then the post editor can keep track of the entity history.

Reply:
Yeh, that could be worth exploring, will have a play around with that idea.

Reply:
Can't we just simplify, if there's history show the document bar, otherwise don't? Alternatively we could also do like the site editor (which we should probably align anyway) and always show the document bar.

Reply:
@SaxonF do you have any views on always showing the document bar in the post editor?